quinn-proto/src/connection/streams/mod.rs

```diff
@@ -271,15 +285,20 @@ impl<'a> SendStream<'a> {
             return Err(WriteError::Blocked);
         }
 
+        stream.write_limit(limit)
+    }
+
+    /// Write bytes under the assumption that `write_limit().unwrap() <= chunk.len()`
+    fn write_unchecked(&mut self, chunk: Bytes) {
```

Comment:
Does this actually need to be unchecked? Seems like it might be fairly cheap to detect incorrect use. Maybe a `debug_assert`, at least...